src/libYARP_sig/include/yarp/sig/PointCloud.h

```diff
@@ -0,0 +1,597 @@
+/*
+ * Copyright (C) 2006-2018 Istituto Italiano di Tecnologia (IIT)
+ * All rights reserved.
+ *
+ * This software may be modified and distributed under the terms of the
+ * BSD-3-Clause license. See the accompanying LICENSE file for details.
+ */
+
+#ifndef YARP_SIG_POINTCLOUD_H
+#define YARP_SIG_POINTCLOUD_H
+
+#include <yarp/sig/Vector.h>
+#include <yarp/sig/PointCloudNetworkHeader.h>
+
+#include <map>
+#include <vector>
+
+
+#include <cstring>
+
+
+namespace yarp {
+    namespace sig {
+        template <class T> class PointCloud;
+        class PointCloudBase;
+    }
+}
+
+// TBD: Copy constructor (needed by compilers such as Intel C++)
```

Comment:
Not sure what you mean here. Can you open a separate issue if this is a problem?

Reply:
@barbalberto is it an issue ?
BTW the copy constructor is defined.